connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -80,7 +81,7 @@ class Column private[sql] (private[sql] val expr: proto.Expression) {
   }
 }
 
-object Column {
+private[sql] object Column {
```

Comment:
Both `Column` and `Column$` are `private[sql]` access scope, so this is not an API for users? 

Seem users cannot create a `Column` in their own package with this pr, for example:

```scala
package org.apache.spark.test

import org.scalatest.funsuite.AnyFunSuite // scalastyle:ignore funsuite
import org.apache.spark.sql.Column

class MyTestSuite
  extends AnyFunSuite // scalastyle:ignore funsuite
{
  test("new column") {
    val a = Column("a") // Symbol apply is inaccessible from this place
    val b = new Column(null) // No constructor accessible from here
  }
```

Reply:
I think `org.apache.spark.sql.Column#apply`  was a public api before. If `private[sql]` is added to `object Column`, it may require more code refactoring work

Reply:
Thanks for your inputs.

Looking the current [Column](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/Column.scala#L35) class, the SQL API give two public APIs to construct the Column:
```
class Column(val expr: Expression) extends Logging {

  def this(name: String) = this(name match {
    case "*" => UnresolvedStar(None)
    case _ if name.endsWith(".*") =>
      val parts = UnresolvedAttribute.parseAttributeName(name.substring(0, name.length - 2))
      UnresolvedStar(Some(parts))
    case _ => UnresolvedAttribute.quotedString(name)
  })
...
```
Right now the client API is very far from completion. We will add new methods in coming PRs. I am sure there will be a `Column(name: String)` for users to use. But it is out the scope of this PR to include all public constructors needed for the client.

The compatibility check added with this PR will grow the check coverage when more and more methods are added in the client. The current check ensures when a new method are added, the new method should be binary compatible with the existing SQL API. When the client API coverage is up (~80%) we can switch to a more aggressive check to ensure we did not miss any methods by mistake.

Reply:
OK, I see what you mean. It seems that this is just an intermediate state. So it really doesn't need to consider users' use now.